app/src/main/java/edu/mills/findeatfood/HttpRequestUtilities.java

```
@@ -0,0 +1,93 @@
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.moshi.MoshiConverterFactory;
+import retrofit2.http.GET;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+/**
+ * Utility class for making HTTP requests to Yummly.
+ * Activities and fragments should only call these methods from an AsyncTask.
+ */
+public final class HttpRequestUtilities {
+
+    private static final String YUMMLY_BASE_URL = "http://api.yummly.com/v1/api/";
+
+    private HttpRequestUtilities() {
+    }
+
+    // Retrofit interfaces to interact with Yummly.
+    private static interface YummlyInterface {
+        //TODO: read keys from private file
+
+        // Request method and URL specified in the annotation.
+        // Callback for the parsed response is the last parameter.
+        @GET("recipe/{recipeId}?_app_id=dac3087f&_app_key=3fe7a0e08b8fc3738d09e764e4f12dcd")
+        Call<Recipe> getRecipe(@Path("recipeId") String recipeId);
+
+        @GET("recipes?_app_id=dac3087f&_app_key=3fe7a0e08b8fc3738d09e764e4f12dcd&maxResult=10")
+        Call<SearchRecipes> searchRecipes(@Query("allowedIngredient") List<String> ingredients,
+                                          @Query("allowedDiet") List<String> diets,
+                                          @Query("allowedAllergy") List<String> allergies,
+                                          @Query("start") int startPosition);
+    }
+
+    /**
+     * Makes a GET request to Yummly to get recipe information.
+     * @param   recipeId the recipe ID to pass to Yummly
+     * @return  the Recipe associated with the passed in recipeId
+     */
+    public static Recipe getRecipe(String recipeId) {
+        YummlyInterface service = createYummlyService(Recipe.class);
+        Call<Recipe> call = service.getRecipe(recipeId);
+        ValContainer<Recipe> recipeContainer = new ValContainer<Recipe>();
+        try {
+            recipeContainer.setVal(call.execute().body());
+        } catch (Exception e) {
```

Don't catch and ignore an exception.